dialects/linux/dproc.c

```diff
@@ -1518,12 +1522,72 @@ process_proc_map(p, s, ss)
 		efs = sv = 1;
 	    else
 		efs = 0;
-	    if (!efs) {
-		if (HasNFS)
-		    sv = statsafely(fp[6], &sb);
-		else
-		    sv = stat(fp[6], &sb);
-	    }
+
+	/*
+	 * Fork a child process and create a pipe.
+	 */
+            if (pipe(pipefd) < 0) {
+                fprintf(stderr, "%s: can't open pipes: %s\n",
+                    Pn, strerror(errno));
+                Exit(1);
+            }
+
+            pid = fork();
+            if (pid < 0) {
+                fprintf(stderr, "%s: can't fork: %s\n",
+                    Pn, strerror(errno));
+                Exit(1);
+            }
+
+            if (pid == 0) {
+	    /*
+	     * Move the child process into the mount namespace of the
+	     * target process before applying stat(2). After that, write
+	     * the return code and stat buffer into pipefd[1].
+	     */
+                int nsfd;
+                char nspath[100];
+
+                close(pipefd[0]);
+                sprintf(nspath, "/proc/%d/ns/mnt", Lp->pid);
+                nsfd = open(nspath, O_RDONLY);
+
+                if (nsfd < 0) {
+                    fprintf(stderr, "%s: can't open ns: %s\n",
+                        Pn, strerror(errno));
+                }
+                setns(nsfd, CLONE_NEWNS);
```

Comment:
This is available since Linux 3.8, should we skip this feature on earlier kernels?

Reply:
Yes, thanks, let me amend it.